src/cargo/util/toml/targets.rs

```diff
@@ -314,34 +359,46 @@ fn clean_tests(
 fn clean_benches(
     toml_benches: Option<&Vec<TomlBenchTarget>>,
     package_root: &Path,
+    edition: Edition,
+    autodiscover: Option<bool>,
     warnings: &mut Vec<String>,
     errors: &mut Vec<String>,
 ) -> CargoResult<Vec<Target>> {
-    let mut legacy_bench_path = |bench: &TomlTarget| {
-        let legacy_path = package_root.join("src").join("bench.rs");
-        if !(bench.name() == "bench" && legacy_path.exists()) {
-            return None;
-        }
-        warnings.push(format!(
-            "path `{}` was erroneously implicitly accepted for benchmark `{}`,\n\
-             please set bench.path in Cargo.toml",
-            legacy_path.display(),
-            bench.name()
-        ));
-        Some(legacy_path)
-    };
+    let mut legacy_warnings = vec![];
+
+    let targets = {
+        let mut legacy_bench_path = |bench: &TomlTarget| {
+            let legacy_path = package_root.join("src").join("bench.rs");
```

Comment:
A nice thing to do in the next PR would be to add `if edition != Edition::Edition2015 { return None; }` over here!

Reply:
ah, this must be one of the instances of "backwards compatibility baggage" you originally referred to in #5330.

sure!